index.bs

```diff
@@ -1468,19 +1512,186 @@ To <dfn>validate a suffix</dfn> |suffix|, run the following steps:
+an explicit {{FilePickerOptions/id}} is also passed in.
+
+For example, given a {{FileSystemDirectoryHandle}} <var>project_dir</var>, the following will show
+a file picker that starts out in that directory:
+
+<pre class=example id="filepickeroptions-starting-directory-example2" highlight=js>
+// The picker will open to the directory of |project_dir| regardless of whether
+// 'foo' has a valid mapping.
+const options = {
+  <l>{{FilePickerOptions/id}}</l>: 'foo',
+  <l>{{FilePickerOptions/startIn}}</l>: |project_dir|,
+};
+</pre>
+
+The {{FilePickerOptions/id}} and {{FilePickerOptions/startIn}} fields control only
+the directory the picker opens to. In the above example, it cannot be assumed that
+the {{FilePickerOptions/id}} 'foo' will map to the same directory as |project_dir|
+once the file picker operation has completed.
+
+Specifying {{FilePickerOptions/startIn}} as a {{WellKnownDirectory}} will result in the dialog
+starting in that directory, unless an explicit {{FilePickerOptions/id}} was also passed
+in which has a mapping to a valid directory.
+
+Below is an example of specifying both an {{FilePickerOptions/id}} and
+{{FilePickerOptions/startIn}} as a {{WellKnownDirectory}}. If there is an existing
+mapping from the given ID to a path, this mapping is used. Otherwise, the path suggested
+via the {{WellKnownDirectory}} is used.
+
+<pre class=example id="filepickeroptions-starting-directory-example3" highlight=js>
+// First time specifying the ID 'foo'. It is not mapped to a directory.
+// The file picker will fall back to opening to the Downloads directory. TODO: link this.
+const options = {
+  <l>{{FilePickerOptions/id}}</l>: 'foo',  // Unmapped.
+  <l>{{FilePickerOptions/startIn}}</l>: <l>{{WellKnownDirectory/"downloads"}}</l>,  // Start here.
+};
+
+// Later...
+
+// The ID 'foo' might or might not be mapped. For example, the mapping for this ID
+// might have been evicted.
+const options = {
+  <l>{{FilePickerOptions/id}}</l>: 'foo',  // Maybe mapped. If so, start here.
+  <l>{{FilePickerOptions/startIn}}</l>: <l>{{WellKnownDirectory/"downloads"}}</l>,  // Otherwise, start here.
+};
+</pre>
+
+</div>
+
+Advisement: The {{FilePickerOptions/startIn}} and {{FilePickerOptions/id}} options were first introduced in Chrome 91.
+
+A user agent holds a <dfn>recently picked directory map</dfn>, which is a
+[=map=] of [=/origins=] to [=path id maps=].
+
+A <dfn>path id map</dfn> is a [=map=] of [=valid path ids=] to paths.
+
+A <dfn>valid path id</dfn> is a [=string=] where each character is [=ASCII alphanumeric=] or "_" or "-".
+
+To prevent a [=path id map=] from growing without a bound, user agents should implement
+some mechanism to limit how many recently picked directories will be remembered. This
+can for example be done by evicting least recently used entries.
+
+The <dfn enum>WellKnownDirectory</dfn> enum lets a website pick one of several well-known
```

Comment:
Minor aside: Should the spec have a note that such WKDs could be virtual, e.g. on OSes that don't expose a filesystem to users but still have a concept of photos storage?

Reply:
Done (sort of).